### DISCUSSION TITLE
can you use OpenMEP with pyRevit?

### DISCUSSION THREAD

**Discussion by jgunstone** [chuongmep/OpenMEP#20] [Q&A]
I imagine that this could be possible but wondering if this has been tried? 
https://www.pyrevitlabs.io/

**Comment by chuongmep** [comment]
Hi @jgunstone , thank you for your interesting with OpenMEP, can I know more what you want to trying to achive with ?
Just provide more information for you in case need to call library is you can call aseembly from library 👍 
https://github.com/chuongmep/OpenMEP/wiki/How-To-Write-Python-Script-With-OpenMEP-Package

**Reply by jgunstone** [reply]
hi - yes I saw this which made me think it will be possible... 
probably best for me to just give it a try and report back. 
[pyRevit](https://github.com/pyrevitlabs/pyRevit) has lots of built in functionality to help quickly produce UI tools in Revit (rather than using dynamo). 
I develop and maintain a pyRevit toolbar suite of automation tools, I'm interested to know whether I could use OpenMEP within that toolbar. 

as an example - here is the imports of a pyRevit script
![image](https://github.com/user-attachments/assets/3d3db1b1-07fe-4b09-8f1b-d8ac7fd56cfa)

**Reply by chuongmep** [reply]
It not much diffrence with the way like Revit Python Shell , you can try with some example like this : 

```python
import clr
path = f"C:\Users\vho2\AppData\Roaming\Dynamo\Dynamo Revit\2.19\packages\OpenMEP\bin\OpenMEPRevit.dll"
clr.AddReferenceToFileAndPath(path)
import OpenMEP
from OpenMEP.Element import Pipe as p
# do something here
```

**Reply by jgunstone** [reply]
thanks - I'll give it a go when I get a moment
